core-bundle/src/Resources/contao/templates/picture/picture_default.html5

```diff
@@ -1,14 +1,20 @@
+<?php $hasSingleAspectRatio = true; ?>
 
 <?php if ($this->sources): ?>
   <picture>
     <!--[if IE 9]><video style="display: none;"><![endif]-->
     <?php foreach ($this->sources as $source): ?>
       <source srcset="<?= $source['srcset'] ?>"<?php if (!empty($source['sizes'])) echo ' sizes="' . $source['sizes'] . '"'; ?><?php if (!empty($source['media'])) echo ' media="' . $source['media'] . '"'; ?>>
+      <?php
+        if (!empty($this->img['width']) && !empty($this->img['height']) && !empty($source['width']) && !empty($source['height']) && abs(($this->img['width'] / $this->img['height']) / ($source['width'] / $source['height']) - 1) > 0.05) {
```

Comment:
```suggestion
        if (true === $hasSingleAspectRatio && !empty($this->img['width']) && !empty($this->img['height']) && !empty($source['width']) && !empty($source['height']) && abs(($this->img['width'] / $this->img['height']) / ($source['width'] / $source['height']) - 1) > 0.05) {
```

This way it will skip the checks if `$hasSingleAspectRation` is already `false`.

Reply:
Not sure that whole calculation should go into the template anyway, wdyt?

Reply:
I would also prefer not to have it in the template.

Reply:
If and how `width`/`height` are set needs to be in the template IMO.
But we could move the single aspect ratio check to the picture factory and pass a `hasSingleAspectRatio` flag to the template. WDYT?

Reply:
👍

Reply:
Done.